src/BugsnagUnity/Native/Android/NativeInterface.cs

```diff
@@ -658,19 +679,24 @@ public void AddMetadata(string section, IDictionary<string,object> metadata)
             {
                 return;
             }
+            using var disposableContainer = new DisposableContainer();
```

Comment:
You need to put the scope in braces or it disposes immediately:

```
using (var disposables = new DisposableContainer()) 
{
  //everything
}
```

Reply:
As discussed I'd forgotten about the `using` _declaration_ which uses the parent scope.